UET/Redpoint.Uet.BuildPipeline/BuildGraph/BuildGraph_Project.xml

```diff
@@ -335,21 +371,33 @@
-          </Agent>
+          <Property Name="CookFlavors" Value="NoCookFlavor" />
+          <Property Name="CookFlavors" Value="$(AndroidGameCookFlavors)" If="'$(TargetPlatform)' == 'Android'" />
+          <ForEach Name="CookFlavor" Values="$(CookFlavors)">
+            <Agent Name="Pak and Stage $(TargetName) $(TargetPlatform) $(TargetConfiguration) $(CookFlavor) (macOS Pak and Stage)" Type="Mac">
+              <Property Name="StageOutputs" Value="#GameStaged_$(TargetName)_$(TargetPlatform)_$(TargetConfiguration)" />
+              <Property Name="StageOutputs" Value="$(StageOutputs)_$(CookFlavor)" If="'$(CookFlavor)' != 'NoCookFlavor'" />
+              <Node Name="Pak and Stage $(TargetName) $(TargetPlatform) $(TargetConfiguration) $(CookFlavor)" Requires="#GameBinaries_$(TargetName)_$(TargetPlatform)_$(TargetConfiguration);#GameCookedContent_$(TargetPlatform)_$(CookFlavor)" Produces="$(StageOutputs)" If="ContainsItem('$(MacPlatforms)', '$(TargetPlatform)', ';')">
+                <Property Name="StagePlatform" Value="$(TargetPlatform)" />
+                <Property Name="StagePlatform" Value="Windows" If="'$(StagePlatform)' == 'Win64'" />
+                <Property Name="StagePlatform" Value="$(StagePlatform)NoEditor" If="(('$(StagePlatform)' == 'Windows') or ('$(StagePlatform)' == 'Mac') or ('$(StagePlatform)' == 'Linux')) and ('$(IsUnrealEngine5)' == 'false')" />
+                <Property Name="CookFlavorArg" Value="" />
+                <Property Name="CookFlavorArg" Value="-cookflavor=$(CookFlavor)" If="'$(CookFlavor)' != 'NoCookFlavor'" />
+                <Property Name="DisableCodeSign" Value="" />
+                <Property Name="DisableCodeSign" Value="-NoCodeSign" If="('$(TargetPlatform)' == 'Win64') or ('$(TargetPlatform)' == 'Mac') or ('$(TargetPlatform)' == 'Linux')" />
+                <Property Name="PackageFlag" Value="" />
+                <Property Name="PackageFlag" Value="-package" If="('$(TargetPlatform)' == 'IOS') or ('$(TargetPlatform)' == 'Android')" />
+                <Command Name="BuildCookRun" Arguments="&quot;-project=$(UProjectPath)&quot; -nop4 $(DisableCodeSign) &quot;-platform=$(TargetPlatform)&quot; $(CookFlavorArg) &quot;-clientconfig=$(TargetConfiguration)&quot; -SkipCook -cook -pak $(PackageFlag) -stage &quot;-stagingdirectory=$(StageDirectory)&quot; -unattended -stdlog -ClientCookedTargets=$(TargetName) -target=$(TargetName)" />
+                <Tag BaseDir="$(StageDirectory)/$(StagePlatform)" Files="..." With="$(StageOutputs)" />
+                <Tag BaseDir="$(ProjectRoot)/Binaries/$(TargetPlatform)" Files="..." With="$(StageOutputs)" If="'$(TargetPlatform)' == 'IOS'" />
+                <Do If="'$(TargetPlatform)' == 'Android'">
+                  <Copy From="$(ProjectRoot)\Binaries\$(TargetPlatform)\..." To="$(ProjectRoot)\Binaries\$(TargetPlatform)_$(CookFlavor)" />
```

Comment:
Should these paths be e.g. `$(ProjectRoot)/Binaries/$(TargetPlatform)/...` for the Mac host nodes?

Reply:
Agree for consistency (copy/paste error, whoops). 

Note: When I was testing earlier (on Windows) and made a mistake, it printed an error with the path `"E:/Project/Root/Path\Binaries\Android\Subfolder/File"` which was kinda weird. I guess BuildGraph internally perhaps always tries to normalize file paths to `/` (which Windows tolerates), but yeah I can't assume that MacOS plays that nice along as well.